src/internal/connector/onedrive/collection.go

```diff
@@ -252,22 +300,49 @@ func (oc *Collection) populateItems(ctx context.Context) {
 				itemSize = itemInfo.OneDrive.Size
 			}
 
-			itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
-				progReader, closer := observe.ItemProgress(ctx, itemData, observe.ItemBackupMsg, itemName, itemSize)
-				go closer()
-				return progReader, nil
-			})
+			if isFile {
+				itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
+					progReader, closer := observe.ItemProgress(ctx, itemData, observe.ItemBackupMsg, itemName+DataFileSuffix, itemSize)
+					go closer()
+					return progReader, nil
+				})
+
+				oc.data <- &Item{
+					id:   itemName + DataFileSuffix,
+					data: itemReader,
+					info: itemInfo,
+				}
+			}
+
+			if itemMeta != nil {
+				// TODO(meain): This right now creates two
+				// progressbars for each file, one for :data and one
+				// for :meta. Should we have some way to show it a
+				// different manner?
+				metaFileSuffix := MetaFileSuffix
+				if !isFile {
+					metaFileSuffix = DirMetaFileSuffix
+				}
+				metaReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
+					progReader, closer := observe.ItemProgress(ctx, itemMeta, observe.ItemBackupMsg, itemName+metaFileSuffix, itemMetaSize)
+					go closer()
+					return progReader, nil
+				})
+
+				oc.data <- &Item{
+					id:   itemName + metaFileSuffix,
+					data: metaReader,
+					info: itemInfo,
+				}
+			}
 
+			// TODO(meain): Should these counts gets updated now that
```

Comment:
How should these counts get updated? Should we consider metadata file as a separate item?

Reply:
Seems reasonable to me.